esp32/modules/setup.py

```diff
@@ -11,7 +11,7 @@ def store_settings(nickname):
 
 def is_developer(nickname):
     if (nickname==""):
-        badge.nvs_set_str('badge', 'setup.state', '2') # Skip the sponsors
+        badge.nvs_set_u8('badge', 'setup.state', 2) # Skip the sponsors
```

Comment:
If we change the type of 'setup.state', we should support backward compatibility / conversion of old values.